lib/src/facts/windows/collection.cc

```diff
@@ -32,13 +33,26 @@ namespace facter { namespace facts {
 
     vector<string> collection::get_external_fact_directories()
     {
-        // Get the user data path
-        TCHAR szPath[MAX_PATH];
-        if (!SUCCEEDED(SHGetFolderPath(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
-            LOG_DEBUG("error finding COMMON_APPDATA: %1%", system_error());
+        if (user::is_admin()) {
+            // Get the common data path
+            TCHAR szPath[MAX_PATH];
+            if (SUCCEEDED(SHGetFolderPath(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
+                path p = path(szPath) / "PuppetLabs" / "facter" / "facts.d";
+                return {p.string()};
+            }
+
+            LOG_WARNING("error finding COMMON_APPDATA, external facts unavailable: %1%", system_error());
+        } else {
+            auto home = user::home_dir();
+            if (!home.empty()) {
+                path p = path(home) / ".puppetlabs" / "opt" / "facter" / "facts.d";
```

Comment:
Sorry, late to this conversation. Yeah, this approach was definitely a compromise between cross-platform (and docs friendly) path consistency vs platform-specific intelligibility (the embedded "opt" is a little weird on *nix, but very weird on windows). For now, I'm okay rolling with this compromise.

If we find it's a source of confusion/aggravation for users (I don't know how extensive the use case for a per-user default external facts path even is), we can always add a sane(r) path later.